File: bintree_java/src/test/TestAvl05.java
package test;


import binarytree.AvlTree;
import binarytree.CandidateRemoval;



public class TestAvl05 extends TestBase {

    @Override
    public void doTask() {
        var avl = new AvlTree<Character>(CandidateRemoval.RIGHT);

        for ( var value : "INFORMATIONTECHNOLOGY".toCharArray() ) {
            System.out.println("\n\n\n\n insert " + value + "\n");
            avl.insert(value);
            displayTree(avl);
        }

        System.out.println();


        for ( var value : "ECI".toCharArray() ) {
            System.out.println("\n\n\n\n remove " + value + "\n");
            avl.remove(value);
            displayTree(avl);
        }

        System.out.println();
    }

}

File: bintree_csharp/test/TestAvl05.cs
using System;
using my.binarytree;


namespace my.test
{
    class TestAvl05 : TestBase
    {
        public override void doTask()
        {
            var avl = new AvlTree<char>(CandidateRemoval.RIGHT);

            foreach (var value in "INFORMATIONTECHNOLOGY")
            {
                Console.WriteLine($"\n\n\n\n insert {value}\n");
                avl.Insert(value);
                DisplayTree(avl);
            }

            Console.WriteLine();


            foreach (var value in "ECI")
            {
                Console.WriteLine($"\n\n\n\n remove {value}\n");
                avl.Remove(value);
                DisplayTree(avl);
            }

            Console.WriteLine();
        }
    }
}
